Assets/Scripts/UI/PickupPup/DogProfile/DogProfile.cs

```diff
@@ -36,6 +36,8 @@ public DogProfileButtonController buttonController
     Image collarImage;
 	[SerializeField]
     UIElement[] descriptionFields; // Normal description must come first (not special)
+    [SerializeField]
+    AffectionMeter affectionMeter;
```

Comment:
While doing this a Null Argument bug kept appearing in this code
had to put the affection meter code above it

public override void SetProfile(Dog dog)
    {
        base.SetProfile(dog);
        affectionMeter.setAffection(dog.Info.Affection);
        **souvenirButton.ToggleEnabled(dog.Info.SouvenirCollected);**
    }